pkg/util/sets/string.go

```diff
@@ -18,43 +18,28 @@ import "sort"
 
 type Set map[string]struct{}
 
-// New returns a new empty Set.
-func New() Set {
-	return make(Set)
-}
-
-// NewWithLength returns an empty Set with the given length.
+// NewWithLength returns an empty Set with the given capacity.
+// It's only a hint, not a limitation.
 func NewWithLength(l int) Set {
 	return make(Set, l)
 }
 
-// NewWith creates a new Set with the given items.
-func NewWith(items ...string) Set {
-	return NewWithLength(len(items)).InsertAll(items...)
-}
-
-// Insert adds the item to the set.
-func (s Set) Insert(item string) Set {
-	s[item] = struct{}{}
-	return s
+// New creates a new Set with the given items.
+func New(items ...string) Set {
```

Comment:
I think the concern was `items ...string` always allocates an array?

Reply:
It doesnot matter, if no args, it is an empty slice. No mem gc at all.

Reply:
It's probably fine. If there is any additional overhead it will be minimal.